distributor/pkg/uniform/watch/uniformwatch.go

```diff
@@ -1,83 +1,114 @@
 package watch
 
 import (
-	"context"
 	"github.com/keptn/go-utils/pkg/api/models"
 	"github.com/keptn/go-utils/pkg/common/retry"
+	"github.com/keptn/keptn/distributor/pkg/config"
 	"github.com/keptn/keptn/distributor/pkg/uniform/controlplane"
+	"github.com/keptn/keptn/distributor/pkg/utils"
 	logger "github.com/sirupsen/logrus"
 	"time"
 )
 
 type IUniformWatch interface {
-	Start(ctx context.Context) string
+	Start(ctx *utils.ExecutionContext) (string, bool)
 }
 
+// UniformWatch periodically checks the control plane api to get information about
+// new subscriptions
 type UniformWatch struct {
+	// HeartbeatInterval is the time duration between each ping
+	// call to the control plane
+	HeartbeatInterval time.Duration
+
+	// MaxHeartbeatRetries determines how often the distributor
+	// shall retry to send a heart beat to the control plane
+	// before giving up
+	MaxHeartbeatRetries int
+
+	// MaxRegisterRetries determines how often the distributor
+	// shall retry to do its initial registration to the control plane
+	// before giving up
+	MaxRegisterRetries int
+
 	controlPlane controlplane.IControlPlane
 	listeners    []SubscriptionListener
-	pingInterval time.Duration
 }
 
-func New(controlPlane controlplane.IControlPlane) *UniformWatch {
+// New creates a new UniformWatch
+// Per default it is configured with: HeartbeatInterval=10s, MaxHeartbeatRetries=20, MaxRegisterRetries=5
+//
+// It returns a pointer to a new UniformWatch without any subscription listeners
+func New(controlPlane controlplane.IControlPlane, env config.EnvConfig) *UniformWatch {
 	return &UniformWatch{
-		controlPlane: controlPlane,
-		pingInterval: 10 * time.Second,
+		HeartbeatInterval:   env.HeartbeatInterval,
+		MaxHeartbeatRetries: env.MaxHeartBeatRetries,
+		MaxRegisterRetries:  env.MaxRegistrationRetries,
+		controlPlane:        controlPlane,
+		listeners:           []SubscriptionListener{},
 	}
 }
 
-func (sw *UniformWatch) Start(ctx context.Context) string {
+// Start triggers starts the attempt(s) to do the initial
+// registration to the control plane.
+// If it was successful, it will start to send heartbeat messages in the background
+// This method does not block
+func (sw *UniformWatch) Start(ctx *utils.ExecutionContext) (string, bool) {
```

Comment:
should we return an error instead of a bool value here? This would make it a little more explicit/easier for the caller to recognize that something went wrong. If we decide to keep the bool return, let's add a short description to the comment above when the returned value is set to false

Reply:
agree, lets do it the `err` way